cppapi/client/eventconsumer.h

```diff
@@ -772,7 +772,16 @@ class DelayedEventSubThread: public omni_thread
 	EventType       et;
 	CallBack        *callback;
 	EventQueue      *ev_queue;
+
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-private-field"
+#endif
 	bool            stateless;
```

Comment:
I could not find a place where this one is used. Maybe clang is right and it can be removed?

Reply:
@mliszcz I've tried removing it, but it is used in the ctor initalizer list, https://github.com/tango-controls/cppTango/pull/605/files/922d4df802b3987d80bad4f669c43b6e482446f0#diff-b8e05bd512a1ddaf88c34d6cb0153bbbL764.

Reply:
Yes, but other than that, it is not used. You can remove it from constructor initializer list as well (and remove constructor parameter `bool _stateless` or wrap it in `TANGO_UNUSED` if it's a public API).

There's no point in storing it if we are not using it.

The parameter could be removed also from call sites, but that is a different issue, out of this PR's scope.

Reply:
> > Yes, but other than that, it is not used. You can remove it from constructor initializer list as well (and remove constructor parameter `bool _stateless` or wrap it in `TANGO_UNUSED` if it's a public API).

I think we can remove from the constructor parameter list. This class is only used internally.

> The parameter could be removed also from call sites, but that is a different issue, out of this PR's scope.

The constructor seems to be called at only one location (in EventConsumer::subscribe_event() in cppapi/client/event.cpp). There is even a test before in this subscribe_event() method ensuring stateless parameter is always true.

Reply:
@mliszcz 

> There's no point in storing it if we are not using it.

Yes of course, but this PR is not about refactoring the code. Nevertheless I've removed the private variable `stateless_` and wrapped the parameter via TANGO_UNUSED as suggested.

Reply:
@bourtemb Our messages crossed. I'll remove the ctor parameter as well.